services/calculate.go

```diff
@@ -167,152 +98,155 @@ func GetResponseHeaderScore(url string) (totalScore int, XSSReportURL string, ma
 		return 0, "", 0, nil, err
 	}
 	var responseHeaderMap map[string]string
+	// Sending Head Request to URL
 	response, err := http.Head(url)
 	if err != nil {
 		return 0, "", 0, nil, err
 	}
 	responseHeaderMap = make(map[string]string)
+	// Constructing Response Header Map
 	for k, v := range response.Header {
 		value := strings.Join(v, ",")
 		responseHeaderMap[k] = value
 	}
-	totalScore = 0
-	var score = 0
-	maxScore = 0
-	if val, ok := responseHeaderMap[XSSHeader]; ok {
-		score, XSSReportURL = GetXSSScore(val)
-	}
-	maxScore = maxScore + 5
-	totalScore = totalScore + score
-	score = 1
-	if val, ok := responseHeaderMap[XFrameHeader]; ok {
-		score = GetXFrameScore(val)
-	}
-	maxScore = maxScore + 5
-	totalScore = totalScore + score
-	score = 2
-	if val, ok := responseHeaderMap[HSTSHeader]; ok {
-		score = GetHSTSScore(val)
-	}
-	maxScore = maxScore + 5
-	totalScore = totalScore + score
-	score = 3
-	if _, ok := responseHeaderMap[CSPHeader]; ok {
-		score = 5
-	}
-	maxScore = maxScore + 5
-	totalScore = totalScore + score
-	score = 3
-	if _, ok := responseHeaderMap[PKPHeader]; ok {
-		score = 5
-	}
-	maxScore = maxScore + 5
-	totalScore = totalScore + score
-	score = 2
-	if val, ok := responseHeaderMap[RPHeader]; ok {
-		score = GetReferrerPolicyScore(val)
-	}
-	maxScore = maxScore + 5
-	totalScore = totalScore + score
-	maxScore = maxScore + 5
-	score = 0
-	if val, ok := responseHeaderMap[XContentTypeHeader]; ok {
-		score = GetXContentTypeScore(val)
-	}
-	totalScore += score
-	maxScore = maxScore + 5
-	if val, ok := responseHeaderMap[Server]; ok {
-		serverInfo = getServerInformation(val)
-	}
-	totalScore += GetHTTPVersionScore(response.Proto)
-	maxScore = maxScore + 5
-	if response.TLS != nil {
-		totalScore += GetTLSVersionScore(response.TLS.Version)
-	}
+	// Calculating Scores for Individual Headers
+	totalScore, maxScore, XSSReportURL = GetXSSScore(responseHeaderMap[XSSHeader], totalScore, maxScore)
```

Comment:
Are we returning the XSSReportURL along with the score response?

Reply:
Yes, but it is not returned to the main function.